### ISSUE TITLE
Deadlock in InitializeAsync 

### ISSUE THREAD

**Issue by huyjack178** [PrismLibrary/Prism#1887]
I have moving code from OnNavigatedTo to InitializeAsync to get data but when starting, my app is hanging

![image](https://user-images.githubusercontent.com/11793001/63576976-902bfe00-c5b7-11e9-856a-82fdb8bf55d4.png)

Error here:
```	
Managed Stacktrace:
=================================================================
	  at <unknown> <0xffffffff>
	  at UIKit.UIApplication:UIApplicationMain <0x002a1>
	  at UIKit.UIApplication:Main <0x000d2>
	  at UIKit.UIApplication:Main <0x00152>
	  at LiveScore.iOS.Application:Main <0x000b2>
	  at <Module>:runtime_invoke_void_object <0x00392>
```

**Comment by nativenolde** [comment]
Hey @brianlagunas I can confirm the deadlock or WSoD as well.

As this is a critical bug, i have to find a workaround first, before i can try to assemble a demo app that reproduces the issue. @huyjack178 it would be great if you also could try to reproduce it...

![Screenshot_1567669940](https://user-images.githubusercontent.com/45496636/64322473-ee4adf00-cfc2-11e9-812f-b25fac7f9726.png)

**Comment by nativenolde** [comment]
@brianlagunas @dansiegel The following demonstration proves the error... I'm still working on a workaround and reproduction sample... It would be great if you could also investigate that as this is a critical issue.

![ezgif-2-df2657c9f47b](https://user-images.githubusercontent.com/45496636/64324854-661b0880-cfc7-11e9-8b9e-b391be30e8c2.gif)

**Comment by nativenolde** [comment]
@brianlagunas @dansiegel @bartlannoeye i was able to reproduce the issue within a minimal viable app... but because that still contains some customer specific artifacts, i cannot share that here on github. Is it possible to share that with you on a private way?

**Comment by dansiegel** [comment]
@nativenolde if you’re a Patron you can share it on Prism Slack privately.

**Comment by nativenolde** [comment]
@dansiegel No, unfortunately not. I'll email you a link to download the app.
As long as "PageViewModel" implements IInitializeAsync, the app not run.
Simply remove the interface to let it run.

**Comment by dansiegel** [comment]
@nativenolde looking at this I have a feeling this is more of an issue that you're not understanding what async/await mean. While you're executing some long running code you're going to see a pause... now if you take some time to update your UI to have some sort of busy indicator when you're navigating then you'll have a better UX... also keep in mind that you do not have to use IInitializeAsync.... there are many valid times where using `async void IInitialize.Initialize` is the better method to use.

**Comment by brianlagunas** [comment]
@nativenolde I'll like to add that all Prism does is await your Task-based code before calling any navigation code. If a dead lock is occurring, it's 99.9% because of your code. That's why we need a reproduction app. We've already tested this feature when we wrote it, and ti worked. So you must be doing something we either didn't think of, or is not correct. Either way, we need a sample.

**Comment by nativenolde** [comment]
@dansiegel You are absolutely right, blocking the navigation process is definitely not a good user experience. This code has been implemented by moving and copying around tons of code to find out what exactly causes the deadlock. I apologize for the confusion.

@brianlagunas I'll email you a link to download a second version of the app, which is a bit more clean. As long as "PageViewModel" implements IInitializeAsync and returns a task wich is not completed, the app will freeze.

```
public async Task InitializeAsync(INavigationParameters parameters)
{
      await Task.Delay(1);
}
```

**Comment by dansiegel** [comment]
@nativenolde you've just demonstrated in that code block that your issue is your code... this is not a bug and is working as designed.

**Comment by brianlagunas** [comment]
For anyone that sees this issue:

**Task != Thread**

If you think that by creating a Task you are automatically processing something on a separate thread, then you are mistaken.
